dbt/include/athena/macros/materializations/models/table/table.sql

```diff
@@ -1,56 +1,111 @@
+-- TODO create a drop_relation_with_versions, to be sure to remove all historical versions of a table
 {% materialization table, adapter='athena' -%}
   {%- set identifier = model['alias'] -%}
 
   {%- set lf_tags = config.get('lf_tags', default=none) -%}
   {%- set lf_tags_columns = config.get('lf_tags_columns', default=none) -%}
   {%- set table_type = config.get('table_type', default='hive') | lower -%}
   {%- set old_relation = adapter.get_relation(database=database, schema=schema, identifier=identifier) -%}
+  {%- set is_ha = config.get('ha', default=false) -%}
+  {%- set s3_data_dir = config.get('s3_data_dir', default=target.s3_data_dir) -%}
+  {%- set s3_data_naming = config.get('s3_data_naming', default='table_unique') -%}
+  {%- set full_refresh_config = config.get('full_refresh', default=False) -%}
+  {%- set is_full_refresh_mode = (flags.FULL_REFRESH == True or full_refresh_config == True) -%}
+  {%- set external_location = config.get('external_location', default=none) -%}
   {%- set target_relation = api.Relation.create(identifier=identifier,
                                                 schema=schema,
                                                 database=database,
                                                 type='table') -%}
+  {%- set tmp_relation = api.Relation.create(identifier=target_relation.identifier ~ '__ha',
+                                             schema=schema,
+                                             database=database,
+                                             s3_path_table_part=target_relation.identifier,
+                                             type='table') -%}
+
+  {%- if (
+    table_type == 'hive'
+    and is_ha
+    and (s3_data_naming in ['table', 'schema_table'] or external_location is not none)
```

Comment:
In the future if we add more naming we might add them here too, and somehow this check can be missed.

How about something like this `'unique' not in s3_data_naming ` as a check?
Doing so we ensure that if we add other naming conventions we have them covered

Reply:
We have the `uuid` naming strategy also 😞 
But yes I am not super fan of this checking ...
A solution could be to call the adapter and then do things in Python like:

```sql
adapter.is_unique_strategy(s3_data_naming)
```

```python
def is_unique_strategy(s3_data_naming: str) -> bool
    naming = S3DataNaming(s3_data_naming)
    return naming.is_unique()
```

Reply:
either that, or maybe change uuid to unique? to be consitent? we are introducing breakcing changes anyway and I believe that almost anybody use uuid

Reply:
Yes it might be worth moving `uuid` to `unique`